js_modules/dagster-ui/packages/ui-core/src/assets/AssetTable.tsx

```diff
@@ -159,24 +157,13 @@ export const AssetTable = ({
           <div style={{flex: 1}} />
           <QueryRefreshCountdown refreshState={refreshState} />
           <Box flex={{alignItems: 'center', gap: 8}}>
-            {checkedAssets.some((c) => !c.definition) ? (
-              <Tooltip content="One or more selected assets are not software-defined and cannot be launched directly.">
-                <Button
-                  intent="primary"
-                  data-testid={testId('materialize-button')}
-                  icon={<Icon name="materialization" />}
-                  disabled
-                >
-                  {checkedAssets.length > 1
-                    ? `Materialize (${checkedAssets.length.toLocaleString()})`
-                    : 'Materialize'}
-                </Button>
-              </Tooltip>
-            ) : (
-              <LaunchAssetExecutionButton
-                scope={{selected: checkedAssets.map((a) => ({...a.definition!, assetKey: a.key}))}}
-              />
-            )}
+            <LaunchAssetExecutionButton
+              scope={{
+                selected: checkedAssets
+                  .filter((a) => !!a.definition)
```

Comment:
You can use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) for this, then you shouldn't need to use the `!` on `a.definition` in the map function below.

```
checkedAssets
  .filter((a): a is Asset => !!a.definition)
  .map((a) => ({...a.definition, assetKey: a.key});
```